### ISSUE TITLE
Many to One in API Seems Obfuscated Now

### ISSUE THREAD

**Issue by heathyates** [dotnet/efcore#2263]
My colleague @WillBaldwin and I contacted a member of the Entity Framework team and they encouraged me to talk here about the recent change from beta3 to beta4 api. Specifically, I feel the following change as shown below is not necessarily for the best. 
# Previous

`builder.Entity<Room>().HasMany(a => a.Books);`
# New

`modelBuilder.Entity<Book>().Collection(c=>c.Books).InverseReference(o=>o.Room).ForeignKey(o => o.RoomId);`

I understand that the team wishes to remove magic and to make a clean break with the EF6 api. However, I humbly suggest this is not it. In my very humble opinion, the new way to implement many to one is obfuscated. This is because the code first approach does not standardize itself with the common nomenclature used in standard relational database design. It also does not flow as well. 

If you have specific reasons for changing this, I would like to know, but at least understand that the additional syntax, in my humble opinion, obfuscates and complicates what used to be straightforward. 

I thank you for your time. I am a huge fan of your team and open source approach.

**Comment by patrickmichalina** [comment]
@heathyates :+1:

**Comment by jdaley** [comment]
One vote for the new names here. I always found the relational modelling language mixed into the entity modelling code a bit hard to follow.

Not sure if you intended it, but your example implies beta4 requires both ends of the relationship and the foreign key to be specified explicitly, whereas beta3 did not and thus had much more concise syntax. I don't think that is the case? The `InverseReference` and `ForeignKey` parts are optional just as they were in beta3.

**Comment by rowanmiller** [comment]
Hey All,

There are varying opinions on our team about the best way to tackle the relationship API and so we decided to try out the current API and see what feedback we get. Thank you for your feedback and we'll continue to monitor this issue and other places to see what folks think.

BTW one of the reasons we liked Reference/Collection was that it's consistent with the ChangeTracker API.

~Rowan
